src/shared/tpm2-util.c

```diff
@@ -391,15 +566,45 @@ static int tpm2_make_primary(
         if (r < 0)
                 return r;
 
+        /* we only need the SRK lock when making the SRK since its not atomic, transient
+         * primary creations don't even matter if they stomp on each other, the TPM will
+         * keep kicking back the same key.
+         */
+        if (use_srk_model) {
+                r = make_lock_file("/run/systemd/tpm2-srk-init", LOCK_EX, &srk_lock);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to take network zone lock: %m");
+        }
+
+        /* Find existing SRK and use it if present */
+        if (use_srk_model) {
+                rc = tpm2_get_srk(c, alg, &alg, primary);
```

Comment:
this is not ideal to me; either the user wants to use the SRK, or they don't. If they do, then it's not up to them what asym alg the SRK is using. So I'm not sure it really makes sense to fail if the user asks for RSA but the SRK is ECC.

Reply:
Sure, ill issue a warning (perhaps notice or info level is better?) and move on. It doesn't matter in the end.

Reply:
Updated in the newest as a warning and moves on.